File: 백준/Silver/11728. 배열 합치기/배열 합치기.java
import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStreamReader;
import java.util.Arrays;
import java.util.StringTokenizer;

public class Main {
	public static void main(String[] args) throws IOException {
		BufferedReader bf = new BufferedReader(new InputStreamReader(System.in));
		StringTokenizer st = new StringTokenizer(bf.readLine());
		StringBuilder sb = new StringBuilder();
		
		int N = Integer.parseInt(st.nextToken());
		int M = Integer.parseInt(st.nextToken());
		
		int[] A = new int[N];
		int[] B = new int[M];
		
		int[] arr = new int[N+M];
		int cnt = 0;
		st = new StringTokenizer(bf.readLine());
		for(cnt=0; cnt<N; cnt++) {
			arr[cnt] = Integer.parseInt(st.nextToken());
		}
		st = new StringTokenizer(bf.readLine());
		for(; cnt<N+M; cnt++) {
			arr[cnt] = Integer.parseInt(st.nextToken());
		}
		
		Arrays.sort(arr);
		
//		System.out.println(Arrays.toString(arr));
		
		for(int i=0; i<arr.length; i++) {
			if(i == arr.length-1) {
				sb.append(arr[i]);				
			}else {
				sb.append(arr[i]).append(" ");
			}
			
		}
		System.out.println(sb);
	}
}

File: 백준/Silver/11728. 배열 합치기/배열 합치기.kt
import java.util.Arrays

fun main(){
    var(N, M) = readln().split(" ").map{ it.toInt() }
    var result = StringBuilder()

    var A = readln().split(" ").toTypedArray().map{ it.toInt() }
    var B = readln().split(" ").toTypedArray().map{ it.toInt() }

    var sortedA = A.sorted()
    var sortedB = B.sorted()



    var a = 0
    var b = 0
    while(a<N && b<M){
        if(sortedA[a] > sortedB[b]){
            result.append(sortedB[b++]).append(" ")
        }else{
            result.append(sortedA[a++]).append(" ")
        }
    }

    if(a == N){
        for(i in b until M){
            result.append(sortedB[i]).append(" ")
        }
    }else{
        for(i in a until N){
            result.append(sortedA[i]).append(" ")
        }
    }

    println(result)
}